x-pack/plugins/enterprise_search/public/applications/app_search/components/documents/search_experience/views/multi_checkbox_facets_view.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import {
+  htmlIdGenerator,
+  EuiCheckboxGroup,
+  EuiFlexGroup,
+  EuiButtonEmpty,
+  EuiSpacer,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+interface Option {
+  value: string;
+  selected: boolean;
+}
+
+interface Props {
+  label: string;
+  options: Option[];
+  showMore: boolean;
+  onMoreClick(): void;
+  onRemove(id: string): void;
+  onSelect(id: string): void;
+}
+
+const getIndexFromId = (id: string) => parseInt(id.split('_')[1], 10);
+
+export const MultiCheckboxFacetsView: React.FC<Props> = ({
+  label,
+  onMoreClick,
+  onRemove,
+  onSelect,
+  options,
+  showMore,
+}) => {
+  const getId = htmlIdGenerator();
+
+  const optionToCheckBoxGroupOption = (option: Option, index: number) => ({
+    id: getId(String(index)),
+    label:
+      option.value ||
+      i18n.translate(
+        'xpack.enterpriseSearch.appSearch.documents.search.multiCheckboxFacetsView.noValue.selectOption',
+        {
+          defaultMessage: '<No value>',
+        }
+      ),
+  });
+
+  const optionToSelectedMapReducer = (
+    selectedMap: { [name: string]: boolean },
+    option: Option,
+    index: number
+  ) => {
+    if (option.selected) {
+      selectedMap[getId(String(index))] = true;
+    }
+    return selectedMap;
+  };
+
+  const checkboxGroupOptions = options.map(optionToCheckBoxGroupOption);
+  const idToSelectedMap = options.reduce(optionToSelectedMapReducer, {});
+
+  const onChange = (checkboxId: string) => {
+    const index = getIndexFromId(checkboxId);
+    const option = options[index];
+    if (option.selected) {
+      onRemove(option.value);
+      return;
+    }
+    onSelect(option.value);
+  };
+
+  return (
+    <>
+      <EuiCheckboxGroup
+        data-test-subj="checkbox_group"
```

Comment:
[not a blocker, just a very minor nit] maybe prefer camelcasing `checkboxGroup` over snake_casing (`_` should mostly be reserved for file names and URLs)

Reply:
https://github.com/elastic/kibana/pull/88764/commits/3da1364701e40e0cf0eb058afe76284058fe7a31